artifactory/services/upload.go

```diff
@@ -138,7 +142,24 @@ func addSymlinkProps(artifact clientutils.Artifact, uploadParams UploadParams) (
 		}
 		artifactProps += utils.ARTIFACTORY_SYMLINK + "=" + artifactSymlink + sha1Property
 	}
-	artifactProps = addProps(uploadParams.GetProps(), artifactProps)
+	props := uploadParams.GetProps()
+	if strings.ContainsAny(props, "build.name") && strings.ContainsAny(props, "build.name") {
```

Comment:
We should find a more aesthetic way to determine whether to add the vcs.url and vcs.revision props.

Reply:
Moved to its own function `GetVcsProps`.